common/JackLoader.cpp

```diff
@@ -0,0 +1,129 @@
+#include <fstream>
+#include "JackLoader.h"
+#include "JackLockedEngine.h"
+
+
+namespace Jack
+{
+
+JackLoader::JackLoader(JackServer* const server) :
+    fServer(server)
+{
+}
+
+int JackLoader::Load(const std::string file)
+{
+    std::ifstream infile(file);
+
+    if (!infile.is_open()) {
+        jack_error("JACK configuration file %s does not exist or cannot be opened for reading.", file.c_str());
+        return -1;
+    }
+
+
+    std::string line;
+    int linenr = -1;
+    while (std::getline(infile, line))
+    {
+        linenr++;
+
+        std::istringstream iss(line);
+
+        char command = 0;
+        if ( !(iss >> command) ) {
+            /* ignoring empty line */
+            continue;
+        }
+
+        switch (command) {
+        case 'l':
+            LoadClient(iss, linenr);
+            break;
+        case 'c':
+            ConnectPorts(iss, linenr);
+            break;
+        case '#':
+            /* ignoring commented lines */
+            break;
+        default:
+            jack_error("JACK configuration file %s line %u contains unkown command '%s'. Ignoring the line!", file.c_str(), linenr, line.c_str());
+            break;
+        }
+    }
+
+    return 0;
+}
+
+void JackLoader::LoadClient(std::istringstream& iss, const int linenr)
+{
+    std::string client_name;
+    if ( !(iss >> client_name) ) {
+        jack_error("Cannot read client name from configuration file line %u '%s'. Ignoring the line!", linenr, iss.str().c_str());
+        return;
+    }
+
+    std::string lib_name;
+    if ( !(iss >> lib_name) ) {
+        jack_error("Cannot read client library name from configuration file line %u '%s'. Ignoring the line!", linenr, iss.str().c_str());
+        return;
+    }
+
+    std::string parameters(iss.str());
+
+
+    /* jackctl_server_load_internal() can not be used
+     * because it calls jack_internal_initialize()
+     * instead of jack_initialize()
+     */
+    int status = 0;
+    int refnum = 0;
+    if (fServer->InternalClientLoad1(client_name.c_str(), lib_name.c_str(), parameters.c_str(), (JackLoadName|JackUseExactName|JackLoadInit), &refnum, -1, &status) < 0) {
+        jack_error("Cannot load client %s from configuration file line %u. Ignoring the line!", client_name.c_str(), linenr);
+        return;
+    }
+
+    if (status & JackFailure) {
+        jack_error("could not load %s, status = 0x%2.0x",
+                 client_name.c_str(), status);
+        return;
+    }
+    if (status & JackNameNotUnique) {
+        jack_error("Internal client name `%s' not unique",
+                 client_name.c_str());
+        return;
+    }
+
```

Comment:
Why not catch all situations where status != 0?

Also, doesn't fServer->InternalClientLoad1 always return false when status != 0?